exampletests/test_address.cpp

```diff
@@ -0,0 +1,113 @@
+#include "yojimbo.h"
+#include "gtest/gtest.h"
+#include "asserts.h"
+
+using namespace yojimbo;
+
+
+TEST(Address, CreateIpv4Direct)
+{
+    Address address( 127, 0, 0, 1, 1000 );
+
+    ASSERT_TRUE( address.IsValid() );
+    ASSERT_EQ( address.GetType(), ADDRESS_IPV4 );
+    ASSERT_EQ( address.GetPort(), 1000 );
+    ASSERT_EQ( address.GetAddress4(), (uint32_t)0x100007f );
+}
+
+TEST(Address, CreateIpv6Direct)
+{
+    const uint16_t addr[] = { 0xFE80, 0x0000, 0x0000, 0x0000, 0x0202, 0xB3FF, 0xFE1E, 0x8329 };
+
+    Address address( addr[0], addr[1], addr[2], addr[2],
+                     addr[4], addr[5], addr[6], addr[7] );
+
+    ASSERT_TRUE( address.IsValid() );
+    ASSERT_EQ( address.GetType(), ADDRESS_IPV6 );
+    ASSERT_EQ( address.GetPort(), 0 );
+    ASSERT_TRUE( isAddress6Match( address.GetAddress6(), addr));
+}
+
+TEST(Address, CreateFromString)
+{
+    Address address( "127.0.0.1" );
+
+    ASSERT_TRUE( address.IsValid() );
+    ASSERT_EQ( address.GetType(), ADDRESS_IPV4 );
+    ASSERT_EQ( address.GetPort(), 0 );
+    ASSERT_EQ( address.GetAddress4(), (uint32_t)0x100007f );
+}
+
+
+TEST(Address, Clear)
+{
+    Address address( 127, 0, 0, 1, 1000 );
+    address.Clear();
+
+    ASSERT_FALSE( address.IsValid() );
+    ASSERT_EQ( address.GetType(), ADDRESS_NONE );
+    ASSERT_EQ( address.GetPort(), 0 );
+    // ASSERT_EQ( address.GetAddress4(), (uint32_t)0x0 );
+}
```

Comment:
This is actually a bug I found while porting these tests. The underlying ipv4_address member isn't cleared?